hypervisor/arch/x86/irq.c

```diff
@@ -322,7 +322,7 @@ void dispatch_interrupt(struct intr_excp_ctx *ctx)
 	uint32_t irq = vector_to_irq[vr];
 	struct irq_desc *desc;
 
-	if (irq == IRQ_INVALID) {
+	if (irq >= NR_IRQS) {
 		goto ERR;
```

Comment:
do we really need such change? As vector_to_irq[irq] will set to IRQ_INVALID for unused one.

Reply:
Here, we assume IRQ_INVALID is >= NR_IRQS. This may be not the case.

Here the irq comes from vector_to_irq tables, it should contain 0--NR_IRQS-1 or INVALID.  Can you add a pre-assumption here?

Reply:
Agree with Jason, this is not necessary.